compiler/p/codegen/GenerateInstructions.cpp

```diff
@@ -314,6 +314,10 @@ TR::Instruction *generateDepLabelInstruction(TR::CodeGenerator *cg, TR::InstOpCo
 TR::Instruction *generateConditionalBranchInstruction(TR::CodeGenerator *cg, TR::InstOpCode::Mnemonic op, bool likeliness, TR::Node * n,
    TR::LabelSymbol *sym, TR::Register *cr, TR::Instruction *preced)
    {
+   // if processor does not support branch hints
+   if (cg->comp()->target().cpu.id() < TR_PPCgp)   
+      return generateConditionalBranchInstruction(cg, op, n, sym, cr);
```

Comment:
You need to include the `preced` argument here in case the caller passed it, otherwise you can get incorrect behaviour.

Reply:
Fixed on 31c3ed6